metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanAspectDao.java

```diff
@@ -69,8 +78,28 @@ public class EbeanAspectDao implements AspectDao, AspectMigrationsDao {
   // more testing.
   private int _queryKeysCount = 375; // 0 means no pagination on keys
 
-  public EbeanAspectDao(@Nonnull final Database server) {
+  /**
+   * Used to control write concurrency when an entity key aspect is present. If a batch contains an
+   * entity key aspect, only allow a single execution per URN
+   */
+  private final LoadingCache<String, Lock> locks;
+
+  public EbeanAspectDao(@Nonnull final Database server, boolean enableLocking) {
     _server = server;
+    if (enableLocking) {
+      this.locks =
```

Comment:
This structure contains a lock per urn which expires after 1 minute. This lock which is used for the write patch is only used when a key aspect is found in a batch. Those batches will also contain all default aspects. These specific batches are serialized using the lock.